### ISSUE TITLE
simulate a real heat pump system with heat storage and two-point control

### ISSUE THREAD

**Issue by jiggey** [CenterHighPerformanceBuildingsPurdue/ACHP#39]
First of all, many thanks for providing the code.
I want to use the cycle model in heat mode to simulate a real heat pump system.
In the real system, the heat pump is either switched on or off according to the temperatures in a heat storage tank.  
Is it possible to simulate a longer period when the heat pump is switched on or off from time to time?

**Comment by abahman** [comment]
Thank you for your question.
Unfortunately, the current ACHP model does not handle dynamic condition. However, the model is based on steady-state. Therefore, you can modify the code to simulate the load every increments too time.

**Comment by jiggey** [comment]
Thank you very much for the quick answer. That sounds like a good plan.

My rough approach would be now:
- Select discrete time interval, how often simulation should run (e.g. every 5 minutes)
- Every 5 minutes in simulation time (when heat pump is switched on):
   - Running the solver in heat mode until it converges (use of converged values as init values in the next iteration)
   - Loading the amount of total heat transfer rate Q_dot of the condenser into my heat storage model (in W/5min). 

Would that make sense? Sorry if I don't quite understand something.

**Comment by abahman** [comment]
Sounds fine with me.
Bare in mind that the condenser is air-typed heat exchanger (HX) in the default cycle model. You can modify the cycle with refrigerant-to-refrigerant internal HX (e.g. Plate heat exchanger) for the heat storage.

**Comment by jiggey** [comment]
Thanks a lot for your help and your hint, I will modify the default cycle model accordingly.

I noticed that with the default cycle model the actual compressor electrical power _W_ of the compressor and the total heat transfer rate _Q_ of the condenser are always the same after the solver has converged:
```
Cycle.PreconditionedSolve()
print('Cycle.Evaporator.Q: ' + str(Cycle.Evaporator.Q)) # 7009.537572462409
print('Cycle.Compressor.W: ' + str(Cycle.Compressor.W)) # 2173.533962381058
Cycle.PreconditionedSolve()
print('Cycle.Evaporator.Q: ' + str(Cycle.Evaporator.Q)) # 7009.537572462409
print('Cycle.Compressor.W: ' + str(Cycle.Compressor.W)) # 2173.533962381058
```
Can this be true? This would mean that it makes no sense to call the solver in discrete intervals for the simulation, if I am only interested in these two values?

**Comment by ibell** [comment]
That's absolutely correct.  The idea of @abahman is for systems in which the _boundary conditions_ are changing, in which case, if you neglect the dynamics of the system, this idea would work well.  If the dynamics of the system itself are important, then this approach would not be suitable.

**Comment by jiggey** [comment]
Thank you, that makes sense. 

Which boundary conditions would typically change over time?

If I neglect these dynamics, I probably can't simulate an inertia in switching the heat pump system on and off or an increase in compressor power consumption as shown in the picture, right?

![heat_pump](https://user-images.githubusercontent.com/18258350/58881463-842c5080-86da-11e9-81e4-2e2af5b132e5.PNG)
